app/views/questionnaire.py

```diff
@@ -224,6 +236,44 @@ def get_thank_you(eq_id, form_type):  # pylint: disable=unused-argument
     return _redirect_to_latest_location(routing_path, collection_id, metadata.get('eq_id'), metadata.get('form_type'))
 
 
+@post_submission_blueprint.route('view-submission', methods=['GET'])
+@login_required
+def get_view_submission(eq_id, form_type):  # pylint: disable=unused-argument
+
+    session_data = get_session_store().session_data
+    g.schema = load_schema_from_params(session_data.eq_id, session_data.form_type)
+
+    if _is_submission_viewable(g.schema.json, session_data.submitted_time):
+        submitted_data = current_app.eq['submitted_responses'].get_item(session_data.tx_id)
+
+        if submitted_data and submitted_data['valid_until'] > datetime.utcnow():
```

Comment:
Do we need to check `submitted_data['valid_until'] > datetime.utcnow()` as it's already checked in the call to `_is_submission_viewable`?

Reply:
No, probably a bit of overkill checking that both times match. I'll remove.